src/helpers/getAssetSettingsFromToml.ts

```diff
@@ -0,0 +1,52 @@
+import { Server } from "stellar-sdk";
+import { getToml } from "methods/getToml";
+import { AssetSupportedActions, AnyObject, AssetType } from "types/types.d";
+
+interface GetAssetSettingsFromToml {
+  assetId: string;
+  networkUrl: string;
+}
+
+export const getAssetSettingsFromToml = async ({
+  assetId,
+  networkUrl,
+}: GetAssetSettingsFromToml): Promise<{
+  homeDomain: string | undefined;
+  supportedActions: AssetSupportedActions | {};
+}> => {
+  const server = new Server(networkUrl);
+
+  // Native (XLM) asset
+  if (assetId === AssetType.NATIVE) {
+    return {
+      homeDomain: undefined,
+      supportedActions: {},
+    };
+  }
+
+  // Other assets
+  let supportedActions: AssetSupportedActions | AnyObject = {};
+  const [, assetIssuer] = assetId.split(":");
+  const accountRecord = await server.loadAccount(assetIssuer);
+  const homeDomain = accountRecord.home_domain;
+
+  if (homeDomain) {
+    const toml = await getToml(homeDomain);
+    const {
+      TRANSFER_SERVER,
+      TRANSFER_SERVER_SEP0024,
+      DIRECT_PAYMENT_SERVER,
+    } = toml;
+
+    supportedActions = {
+      sep6: Boolean(TRANSFER_SERVER),
+      sep24: Boolean(TRANSFER_SERVER_SEP0024),
+      sep31: Boolean(DIRECT_PAYMENT_SERVER),
+    };
```

Comment:
From a UX perspective, its not ideal to have SEP-24 deposit and withdraw displayed as a possible action for an asset if it is not actually enabled. It might make sense to check the info endpoints of each transfer server and verify that the anchor supports each action for the `assetId` here instead of calling `checkInfo()` after the action has been selected.

Reply:
If we do all these checks on load, it will be a lot of information to process for the user (if there are multiple incomplete actions, for example) if we show logs for what is failing. If something is wrong with a specific action, it's easier to find the issue. What do you think?